shell/edit/auth/azuread.vue

```diff
@@ -190,6 +190,23 @@ export default {
       }
     },
 
+    determineEndpointKeyType(endpointTypes) {
+      let out = null;
+
+      for ( const [key, values] of Object.entries(endpointTypes) ) {
+        const mappedValues = Object.values(values).map(v => v.replace(TENANT_ID_TOKEN, this.model?.tenantId));
+        const valuesToCheck = Object.keys(values).map(k => this.value[k]);
+
+        if ( mappedValues === valuesToCheck ) {
+          out = key;
+        } else {
+          out = 'custom';
+        }
+      }
+
+      return out;
+    },

```

Comment:
Arrays can't be properly compared using `===` try lodash:
```
import isEqual from 'lodash/isEqual'

...

if(isEqual(mappedValues, valuesToCheck))

...

```

I think you also need to change where in this process you're setting `out = 'custom' `. As it is now, this loop is setting `out = 'custom'` for each region (standard, china, custom) that doesn't correspond to the current config. That is, if the first region checked is the one we want, out is gonna be re-set to 'custom' when the next region is iterated over. Try using 'custom' as the initial value for `out` rather than setting it within the loop.

This is, admittedly, nitpicky, but I find this loop hard to read: I think some more specific variable names would be helpful here.

Reply:
Great points, makes a lot of sense to set `out` as `custom` initially. Also changed the equality checking to use lodash and renamed a few of the variables.

Reply:
It's easier for me to read now, thank you :)